### PR TITLE
Fix negated is na

### PR DESCRIPTION
**Pull request by edgararuiz**
The new solution uses `~` for SELECT clauses and `NOT` for WHERE clauses.  It uses the `sql_currentcontext()` function to switch between the two.  

Here are the tests that worked: 

```
db_mtcars %>%
  mutate(x = !is.na(mpg))

db_mtcars %>%
  arrange(!is.na(mpg))

db_mtcars %>%
  group_by(!is.na(mpg))  %>%
  tally()

db_mtcars %>%
  filter(!is.na(mpg))

db_mtcars %>%
  filter(am != 0)

db_mtcars %>%
  filter(!am == 0)

db_mtcars %>%
  filter(!mpg < 30 )

db_mtcars %>%
  filter(!mpg < 30 | !cyl < 8 )

db_mtcars %>%
  filter(!mpg >= 30 )

db_mtcars %>%
  filter(!mpg == 21 )

db_mtcars %>%
  filter(mpg != 21 )
```

These didn't 
```
db_mtcars %>%
  mutate(x = am != 0) %>%
  show_query()

db_mtcars %>%
  mutate(x = (am == 0)) %>%
  show_query()
```

### CODE REVIEW

**Comment by edgararuiz** [comment]
PTAL @hadley

File: R/db-odbc-mssql.R [hunk 1]
```diff
@@ -135,10 +138,27 @@
...
   if (context$clause %in% c("SELECT", "ORDER")) {
-    sql_expr(convert(BIT, iif(UQ(x) %is% NULL, 1L, 0L)))
+    sql_expr(convert(BIT, !!x))
   } else {
     sql_expr( ((!!x) %is% NULL) )
   }
 }
 
+mssql_not_sql_prefix <- function() {
+  function(...) {
```
Comment (hadley) [inline R/db-odbc-mssql.R]: Not only has one argument, so you can simplify this a bit

**Comment by edgararuiz** [comment]
- Added two new custom `sql_infix()`, one for the operators and one for AND/OR

Here are the new tests that passed:

```

db_mtcars %>%
  mutate(x = mpg <= 20 & am == 0) 


db_mtcars %>%
  mutate(x = !(mpg <= 20) & !(am == 0)) 

db_mtcars %>%
  mutate(x = mpg <= 20 | am == 0) 


db_mtcars %>%
  mutate(x = !(mpg <= 20) | !(am == 0)) 

db_mtcars %>%
  mutate(x = (am != 0))

db_mtcars %>%
  mutate(x = (am == 0))

db_mtcars %>%
  mutate(x = (mpg > 20)) 

db_mtcars %>%
  mutate(x = (mpg < 20)) 

db_mtcars %>%
  mutate(x = (mpg >= 20)) 

db_mtcars %>%
  mutate(x = (mpg <= 20)) 

db_mtcars %>%
  mutate(x = !(am != 0))

db_mtcars %>%
  mutate(x = !(am == 0))

db_mtcars %>%
  mutate(x = !(mpg > 20)) 

db_mtcars %>%
  mutate(x = !(mpg < 20)) 

db_mtcars %>%
  mutate(x = !(mpg >= 20)) 

db_mtcars %>%
  mutate(x = !(mpg <= 20)) 
```

**Comment by edgararuiz** [comment]
PTAL @hadley

**Comment by edgararuiz** [comment]
PTAL @hadley

**Comment by hadley** [comment]
Can you please update the bullet point in `NEWS.md`? (Near the bottom under database specific translations)

**Pull request merged**